servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -287,14 +302,40 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge)
           .merge(
               toHash(merge.getFromRefName(), merge.getFromHash(), true).get(),
               BranchName.of(branchName),
-              toHash(hash, true));
+              toHash(hash, true),
+              getResetMergePropsFunc(MergeOp.MERGE, merge.getFromRefName()));
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(e.getMessage(), e);
     } catch (ReferenceConflictException e) {
       throw new NessieConflictException(e.getMessage(), e);
     }
   }
 
+  private BiFunction<Serializer<CommitMeta>, ByteString, ByteString> getResetMergePropsFunc(
+      MergeOp mergeOp, String fromRefName) {
+    return (ser, in) -> {
+      try {
+        CommitMeta commitMeta = ser.fromBytes(in);
+        String currUser =
```

Comment:
This whole assignment could be externalized to a separate method to match the behavior of the commit-code-path.
I.e. both should do the same thing for setting the committer + commitTime.

Reply:
Can you please elaborate, this is already an abstracted method, referred from both transplant and merge.

Reply:
Unify with the code-path for the commit-operation + merge + transplant.